protocol/xmrmaker/swap_state.go

```diff
@@ -165,6 +168,104 @@ func newSwapStateFromStart(
 	return s, nil
 }
 
+// checkIfAlreadyClaimed returns true if the ETH has already been
+// claimed by us, false otherwise.
+func checkIfAlreadyClaimed(
+	b backend.Backend,
+	ethSwapInfo *db.EthereumSwapInfo,
+) (bool, error) {
+	// check if swap actually completed and we didn't realize for some reason
+	// this could happen if we restart from an ongoing swap
+	contract, err := contracts.NewSwapFactory(ethSwapInfo.ContractAddress, b.ETHClient().Raw())
+	if err != nil {
+		return false, err
+	}
+
+	stage, err := contract.Swaps(b.ETHClient().CallOpts(b.Ctx()), ethSwapInfo.SwapID)
+	if err != nil {
+		return false, err
+	}
+
+	switch stage {
+	case contracts.StageInvalid:
+		// this should never happen
+		return false, fmt.Errorf("%w: contract swap ID: %s", errSwapDoesNotExist, ethSwapInfo.SwapID)
+	case contracts.StageCompleted:
+		// check if we already claimed, or if the swap was refunded
+	case contracts.StagePending, contracts.StageReady:
+		return false, nil
+	default:
+		panic("Unhandled stage value")
+	}
+
+	filterQuery := ethereum.FilterQuery{
+		FromBlock: ethSwapInfo.StartNumber,
+		Addresses: []ethcommon.Address{ethSwapInfo.ContractAddress},
+	}
+
+	claimedTopic := common.GetTopic(common.ClaimedEventSignature)
+
+	// let's see if we have logs
+	logs, err := b.ETHClient().Raw().FilterLogs(b.Ctx(), filterQuery)
+	if err != nil {
+		return false, fmt.Errorf("failed to filter logs for topic %s: %s", claimedTopic, err)
+	}
+
+	log.Debugf("filtered for logs from block %s to head", filterQuery.FromBlock)
+
+	var foundClaimed *ethtypes.Log
```

Comment:
This is declared as `*ethtypes.Log`, but the code is only using it as a boolean.  You could declare `claimFound := false` here and just set it to `true` when you find it, and then you won't need the nil check at the end.